src/vector/__init__.py

```diff
@@ -5,7 +5,6 @@
 
 import typing
 
-from vector._backends.awkward_ import VectorAwkward
```

Comment:
This line was a last-minute mistake: it was designed up to this point to not import `vector._backends.awkward_` at startup, only `vector._backends.numpy_`. I must have forgotten that in some later step.

Would it be preferable to have `VectorAwkward` be undefined, rather than `None`, if `awkward` can't be imported? That's more in line with what I'd expect.

Independently of that, it may be better to do the test for importability like this:

```python
try:
    import awkward
except ModuleNotFoundError:
    pass   # or VectorAwkward = None
else:
    from vector._backends.awkward_ import VectorAwkward
```

That way, difficulties importing `awkward` are distinguishable from difficulties importing `vector._backends.numpy_`, following the "do as little as possible in a `try` block" rule. Specifically, it's quite normal for `awkward` to not be installed on a given system, but assuming that it is installed, being unable to import one of Vector's own submodules would be some kind of bug.

Reply:
The problem with not having defined at all is `__all__`, which probably should not be dynamic (we certainly would have to trick MyPy if it was dynamic, as mypy needs to know if `vector.VectorAwkward` exists without knowing if awkward is installed; flake8 likely would be unhappy too, etc). I can experiment with making it dynamic and see what happens. Removing it from `__all__` at runtime would make `__dir__ `work better, so maybe that is better.

This is at the end, so it _should_ be pretty safe, but I still like your version above better.

Reply:
MyPy is happy, but flake8 blows up.

Reply:
Implemented an intermediate solution. It's still set to `None`, but doesn't show up in `dir()` / IPython (Python 3.7+ only). Also added "simple" tests to make sure it works without Awkward / Numba installed.